server/authorizer/authorizer.go

```diff
@@ -277,3 +282,20 @@ func GetBodyOnContext(ctx context.Context) (interface{}, bool) {
 	}
 	return input.parsed, true
 }
+
+// Note(philipc): Copied over from server/server.go
+func readPlainBody(r *http.Request) (io.ReadCloser, error) {
+	if strings.Contains(r.Header.Get("Content-Encoding"), "gzip") {
+		gzReader, err := gzip.NewReader(r.Body)
+		if err != nil {
+			return nil, err
+		}
+		bytesBody, err := io.ReadAll(gzReader)
+		if err != nil {
+			return nil, err
+		}
+		defer gzReader.Close()
```

Comment:
If `io.ReadAll()` fails, then `gzReader` will be left open. I think that could leak?

Reply:
Yep, that looks like a potential bug, straight from copy/pasting the original. I'll fix it presently.